File: Conversion/Solution1.java
/* 
	Solution 1 
	
	Each 1 in the XOR represents a bit that is different between A and B. Therefore, to check the
	number of bits that are different between A and B, we simply need to count the number of bits
	in A^B that are 1.
*/


/* 
	Time Complexity: 
	Space Complexity: 
*/

int bitSwapRequired(int a, int b){
	int count = 0;
	for(int c = a ^ b; c != 0; c = c >>> 1){
		count += c & 1;
	}
	return count;
}

File: Hamming_Distance/Solution1.cs
/* 
	Solution 1 
	
	While either input integer is greater than 0
		Add one to the total distance if the first bit is different for both integers
		Then remove the first bit for both integers

	Return the total distance
*/


/* 
	Time Complexity: O(n), where n is the number of bits in the larger of two integers 
	Space Complexity: O(1)
*/

public class Solution {
    public int HammingDistance(int x, int y) {
        int hDist = 0;
        while(x > 0 || y > 0){
            hDist += ((x & 1) ^ (y & 1));
            x >>= 1;
            y >>= 1;
        }
        
        return hDist;
    }
}

/*
	LEET CODE RESULTS

	Runtime: 32 ms, faster than 99.50% of C# online submissions for Hamming Distance.
	Memory Usage: 15.2 MB, less than 39.70% of C# online submissions for Hamming Distance.
*/